cpp/include/raft/neighbors/detail/cagra/graph_core.cuh

```diff
@@ -160,7 +140,7 @@ __global__ void kern_sort(
     uint32_t next_mask = mask << 1;
```

Comment:
I am wondering if RAFT's [bitonic sort](https://github.com/rapidsai/raft/blob/cc4a76ba19b4b01ca5cbf454365f5f48f4bf313c/cpp/include/raft/util/bitonic_sort.cuh#L43) method can be utilized here. (If yes, then it should probably be a separate PR).

Reply:
The bitonic sort method can be partially used in the `kern_sort` function, specifically the following part.
https://github.com/rapidsai/raft/blob/1d1c5234f6d6b43f14f51a269c186f4b0ee1233a/cpp/include/raft/neighbors/detail/cagra/graph_core.cuh#L180-L186
Is it better to use the RAFT method instead of the current method?
(I think the current one is also fine because this part will not be a performance bottleneck, and it is consistent with the rest of the function.)

Reply:
For this PR it is fine to keep the current method. We shall return to this question later. The goal is improve maintainability of the codebase, by reusing existing methods. But if the existing method fits our purpose only partially, then we should think over what is the cleanest solution.